### PR TITLE
types/views: add generic Slice[T] and remove StringSlice

### PR DESCRIPTION
**Pull request by maisem**
Also make IPPrefixSliceOf use Slice[netaddr.IPPrefix] as it also
provides additional functions besides the standard ones provided by
Slice[T].

I considered making `type StringSlice = Slice[string]` but decided against it.

Signed-off-by: Maisem Ali <maisem@tailscale.com>

### CODE REVIEW

File: types/views/views.go [hunk 1]
```diff
@@ -44,83 +44,77 @@ func (v *StringSlice) UnmarshalJSON(b []byte) error {
...
-func (v IPPrefixSlice) IsNil() bool { return v.ж == nil }
+func (v IPPrefixSlice) IsNil() bool { return v.ж.IsNil() }
 
 // Len returns the length of the slice.
-func (v IPPrefixSlice) Len() int { return len(v.ж) }
+func (v IPPrefixSlice) Len() int { return v.ж.Len() }
 
-// At returns the IPPrefix at index `i` of the slice.
-func (v IPPrefixSlice) At(i int) netaddr.IPPrefix { return v.ж[i] }
+// At returns the string at index `i` of the slice.
```
Comment (bradfitz) [inline types/views/views.go:79]: why'd this change to "string"?

Comment (maisem) [reply types/views/views.go:79]: that's not intentional, I fixed it locally but then got distracted by the `staticcheck` and `go generate` failures

File: types/views/views.go [hunk 2]
```diff
@@ -44,83 +44,77 @@ func (v *StringSlice) UnmarshalJSON(b []byte) error {
...
+func (v Slice[T]) AsSlice() []T {
 	return v.AppendTo(v.ж[:0:0])
 }
 
 // IPPrefixSlice is a read-only accessor for a slice of netaddr.IPPrefix.
 type IPPrefixSlice struct {
-	// It is named distinctively to make you think of how dangerous it is to escape
-	// to callers. You must not let callers be able to mutate it.jd
-	ж []netaddr.IPPrefix
+	ж Slice[netaddr.IPPrefix]
```
Comment (bradfitz) [inline types/views/views.go:67]: Did you try embedding this so you wouldn't need all the forwarding accessors like IsNil, Len, At, etc?

**Review by bradfitz — APPROVED** [review]

File: types/views/views.go (hunk 2 above)
Comment (maisem) [reply types/views/views.go:67]: I did, didn't work.

**Comment by bradfitz** [comment]
I'm fine disabling the "go generate" and "staticcheck" CI actions for a week or two.

File: types/views/views.go (hunk 2 above)
Comment (bradfitz) [reply types/views/views.go:67]: Sad. Oh well.

**Comment by maisem** [comment]
`go generate` failure looks related to https://github.com/golang/go/issues/51629
`staticcheck` failure looks related to https://github.com/dominikh/go-tools/issues/1166

**Comment by maisem** [comment]
> I'm fine disabling the "go generate" and "staticcheck" CI actions for a week or two.

I disabled the workflows from the UI

**Comment by maisem** [comment]
Tracking issue to re-enable https://github.com/tailscale/tailscale/issues/4194

**Pull request merged**